backend/ee/danswer/server/query_and_chat/chat_backend.py

```diff
@@ -122,3 +134,117 @@ def handle_simplified_chat_message(
         response.answer_citationless = remove_answer_citations(answer)
 
     return response
+
+
+# take in a list of previous chat messages
+# do query rephrasing like the other endpoint
+@router.post("/send-message-simple-with-history")
+def handle_simplified_chat_message_2(
+    req: BasicCreateChatMessageWithHistoryRequest,
+    user: User | None = Depends(current_user),
+    db_session: Session = Depends(get_session),
+) -> ChatBasicResponse:
+    """This is a Non-Streaming version that only gives back a minimal set of information"""
+
+    if len(req.messages) == 0:
+        raise HTTPException(status_code=400, detail="Messages cannot be zero length")
+
+    query = req.messages[-1].message
+    if not query:
+        raise HTTPException(status_code=400, detail="Empty chat message is invalid")
```

Comment:
we may also want to verify the correctness of the passed in message chain (e.g. that it starts with a user message and ends with a user message + alternates between user/assistant throughout)

Reply:
> this could probably check all messages to ensure they're not empty?

Would the idea to be to fail if any message is empty?

> we may also want to verify the correctness of the passed in message chain (e.g. that it starts with a user message and ends with a user message + alternates between user/assistant throughout)

Done.

Reply:
Yep that's correct. Any empty message is invalid. This could be a follow up item (not too important, could also be a part of the "API refactor"